chapters/embeddings.qmd

```diff
@@ -0,0 +1,646 @@
+  scale_color_brewer(palette = "Dark2")
+
+orig_p <-
+  base_p +
+  geom_point(
+    pch = 16,
+    cex = 8,
+    fill = "white",
+    col = "white",
+    show.legend = FALSE
+  ) +
+  geom_point(
+    pch = 1,
+    cex = 8,
+    aes(col = species),
+    fill = "white",
+    show.legend = FALSE
+  ) +
+  geom_text(aes(label = sample, col = species), show.legend = FALSE) +
+  labs(title = "(a) Original Data", x = "Predictor A", y = "Predictor B")
+
+### 
+
+knn_p <- base_p
+
+for(i in 1:n) {
+  tmp <- pen_subset[nearest[[i]], 1:2] %>% rename_with(~paste0(., "_end"))
+  dat <- pen_subset %>% slice(rep(i, 2)) %>% bind_cols(tmp)
+
+  knn_p <-
+    knn_p +
+    geom_segment(data = dat, aes(xend = length_end, yend = depth_end),
+                 col = "grey")
+}
+knn_p <-
+  knn_p +
+  geom_point(
+    pch = 16,
+    cex = 8,
+    fill = "white",
+    col = "white",
+    show.legend = FALSE
+  ) +
+  geom_point(
+    pch = 1,
+    cex = 8,
+    aes(col = species),
+    fill = "white",
+    show.legend = FALSE
+  ) +
+  geom_text(aes(label = sample, col = species), show.legend = FALSE) +
+  labs(title = "(b) 2-Nearest Neighbors", x = "Predictor A", y = "Predictor B") 
+
+### 
+
+  one_d_p <-
+    one_d_df %>%
+    ggplot(aes(x = MDS_1, y = method)) +
+    geom_point(
+      pch = 16,
+      cex = 8,
+      fill = "white",
+      col = "white",
+      show.legend = FALSE
+    ) +
+    geom_point(
+      pch = 1,
+      cex = 8,
+      aes(col = species),
+      fill = "white",
+      show.legend = FALSE
+    ) +
+    geom_text(aes(label = sample, col = species), show.legend = FALSE) +
+    labs(title = "(c) 1 Dimensional MDS (two ways)", y = NULL, 
+         x = "Embedded Value") +
+    scale_color_brewer(palette = "Dark2")
+  
+### 
+
+layout <- "
+AAAAAAAA
+AAAAAAAA
+AAAAAAAA
+AAAAAAAA
+AAAAAAAA
+AAAAAAAA
+#BBBBBB#
+#BBBBBB#
+"
+
+((orig_p + knn_p) / one_d_p) +
+  plot_layout(design = layout)
+```
+
+There are many different MDS methods. @Ghojogh2023 has an excellent review of them and their nuances. 
+
+Some MDS methods compute all pairwise distances between points and use this as the input to the embedding algorithm. This is similar to how PCA can be estimated using the covariance or correlation matrices. One technique, _Non-Metric MDS_ [@kruskal1964multidimensional;@kruskal1964nonmetric;@sammon1969nonlinear], finds embeddings that minimize an objective function called "stress":
+
+$$
+\text{Stress} = \sqrt{\frac{\sum\limits^{n_{tr}}_{i = 1}\;\sum\limits^{n_{tr}}_{j = i+1}\left(d(x_i, x_j) - d(x^*_i, x^*_j)\right)^2}{\sum\limits^{n_{tr}}_{i = 1}\;\sum\limits^{n_{tr}}_{j = i+1}d(x_i, x_j)^2}}
+$$
+
+The denominator uses the squared difference between the pairwise distances in the original values ($x$) and the smaller embedded dimension ($x^*$). The summations only move along one diagonal of the distance matrices to reduce redundant computations. @fig-mds-example(c, top row) has the resulting one dimensional projection of our two-dimensional data.
```

Comment:
I think denominator should be numerator?  Also, I'm not following the sentence "The summations only move along one diagonal..."  The formula as written sums across all pairs of points.